### ISSUE TITLE
The first synchronization is very slow. Is there any optimization strategy?

### ISSUE THREAD

**Issue by Haley-Wong** [matrix-org/matrix-ios-sdk#561]
I have an account, there are more than 1000 rooms, written after re-installation, initialization synchronization, to perform more than 10 minutes, how to optimize this problem?

I did a test. When there are 100 rooms, the initialization synchronization is about 5 seconds. When there are 600 rooms, the initialization synchronization is about 30 seconds. When there are 1100 rooms, the initialization synchronization is about 10 minutes. . When the number of rooms increases, the initial synchronization will be exponentially long. Should you consider how to optimize this time?

**Comment by manuroe** [comment]
Yes, we are currently working on improving the initial sync (time & data usage): https://github.com/vector-im/riot-ios/issues/1931.

This significantly improves the initial sync time if you are in rooms with thousands of users.

**Comment by manuroe** [comment]
> When the number of rooms increases, the initial synchronization will be exponentially long.

@Haley-Wong, as you are the 1st to identify this problem, could you also send some logs that demonstrate this exponential slowness issue, please?
